desc/objectives/_coils.py

```diff
@@ -1599,3 +1612,211 @@ def compute(self, params, constants=None):
+        REGCOIL algorithm described in [1]_ .
+
+    """
+
+    weight_str = (
+        "weight : {float, ndarray}, optional"
+        "\n\tWeighting to apply to the Objective, relative to other Objectives."
+        "\n\tMust be broadcastable to to Objective.dim_f"
+        "\n\tWhen used with QuadraticFlux objective, this acts as the regularization"
+        "\n\tparameter (with w^2 = lambda), with 0 corresponding to no regularization."
+        "\n\tThe larger this parameter is, the less complex the surface current will "
+        "be,\n\tbut the worse the normal field."
+    )
+    __doc__ = __doc__.rstrip() + collect_docs(
+        target_default="``target=0``.",
+        bounds_default="``target=0``.",
+        loss_detail=" Note: has no effect for this objective.",
+        overwrite={"weight": weight_str},
+    )
+
+    _coordinates = "tz"
+    _units = "A/m"
+    _print_value_fmt = "Surface Current Regularization: "
+
+    def __init__(
+        self,
+        surface_current_field,
+        target=None,
+        bounds=None,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        loss_function=None,
+        deriv_mode="auto",
+        source_grid=None,
+        name="surface-current-regularization",
+    ):
+        from desc.magnetic_fields import (
+            CurrentPotentialField,
+            FourierCurrentPotentialField,
+        )
+
+        if target is None and bounds is None:
+            target = 0
+        assert isinstance(
+            surface_current_field, (CurrentPotentialField, FourierCurrentPotentialField)
+        ), (
+            "surface_current_field must be a CurrentPotentialField or "
+            + f"FourierCurrentPotentialField, instead got {type(surface_current_field)}"
+        )
+        self._surface_current_field = surface_current_field
+        self._source_grid = source_grid
+
+        super().__init__(
+            things=[surface_current_field],
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            loss_function=loss_function,
+            deriv_mode=deriv_mode,
+            name=name,
+        )
+
+    def build(self, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        from desc.magnetic_fields import FourierCurrentPotentialField
+
+        surface_current_field = self.things[0]
+        if isinstance(surface_current_field, FourierCurrentPotentialField):
+            M_Phi = surface_current_field._M_Phi
+            N_Phi = surface_current_field._N_Phi
+        else:
+            M_Phi = surface_current_field.M + 1
+            N_Phi = surface_current_field.N + 1
```

Comment:
Just curious: why +1 as the default instead of +0?

Reply:
If there was a reason, I do not remember tbh. The surface resolution is hard to tie to the necessary resolution for the current potential, especially given that the function for the current potential is unknown when it is a CurrentPotentialField. I could maybe make this more conservative, but for now this is not really used

Reply:
If there is not a good reason for this choice, I would recommend changing this default to be simply `M_Phi = surface_current_field.M`. Because hard-coding a "magic number" like `+ 1` makes it seem like there was a reason that is not documented.